canvas/cli/common.py

```diff
@@ -1,12 +1,19 @@
+import json
 import sys
 
 # keys: [(items key, title, pretty title), ...]
 def cli_list(items, keys, table = False, skip_headers = False,
-        collective_name = 'items', sort_key = 'id'):
+        collective_name = 'items', sort_key = 'id',
+        **kwargs):
     if (len(items) == 0):
         print("No %s found." % (collective_name), file = sys.stderr)
         return 0
 
+    is_json = kwargs.get('json', False)
+    if (is_json):
+        print(json.dumps(items, indent = 4, sort_keys = True))
+        return 0
```

Comment:
This seems a little too early.
We essentially want a JSON version (list of objects) of what is output with `--table`,
and this includes some cleaning done in the loop below.